src/web/app/components/adding-question-panel/adding-question-panel.component.spec.ts

```diff
@@ -36,4 +36,42 @@ describe('AddingQuestionPanelComponent', () => {
   it('should create', () => {
     expect(component).toBeTruthy();
   });
+
+  it('raises the selected event when templateQuestionModalEvent is invoked', () => {
+    spyOn(component.templateQuestionModalEvent, 'emit');
+
+    const button: any = fixture.nativeElement.querySelector('button');
+
+    button.click();
+    fixture.detectChanges();
+
+    component.templateQuestionModalHandler();
+
+    expect(component.templateQuestionModalEvent.emit).toHaveBeenCalled();
+  });
+
+  it('raises the selected event when populateAndShowNewQuestionFormEvent is invoked', () => {
+    const type: FeedbackQuestionType = FeedbackQuestionType.MCQ;
+    const button: any = fixture.nativeElement.querySelector('button');
+
+    spyOn(component.populateAndShowNewQuestionFormEvent, 'emit');
+    button.click();
+    fixture.detectChanges();
+
+    component.populateAndShowNewQuestionFormHandler(FeedbackQuestionType.MCQ);
+
+    expect(component.populateAndShowNewQuestionFormEvent.emit).toHaveBeenCalledWith(type);
+  });
+
+  it('raises the selected event when templateQuestionModalEvent is invoked', () => {
+    const button: any = fixture.nativeElement.querySelector('button');
+
+    spyOn(component.copyQuestionsFromOtherSessionsEvent, 'emit');
+    button.click();
+    fixture.detectChanges();
+
+    component.copyQuestionsFromOtherSessionsHandler();
+
+    expect(component.copyQuestionsFromOtherSessionsEvent.emit).toHaveBeenCalled();
+  });
 });
```

Comment:
these 3 tests seem very similar to each other, maybe we can use a for loop and compress them into 1 single test